include/simics/dmllib.h

```diff
@@ -599,6 +599,423 @@ _identity_eq(const _identity_t a, const _identity_t b) {
     return a.id == b.id && a.encoded_index == b.encoded_index;
 }
 
+
+typedef set_error_t (*_deserializer_t)(attr_value_t *val, void *addr);
+typedef void (*_serializer_t)(const void *addr, attr_value_t *val);
+
+typedef struct {
+    uint32      *indices;
+    void        *args;
+    _identity_t *domains;
+    uint64      no_domains;
+} _simple_event_data_t;
+
+UNUSED static int
+_simple_event_predicate(lang_void *_data, lang_void *match_data) {
+    ASSERT(match_data);
+
+    if (!_data) {
+        return 0;
+    }
+
+    _simple_event_data_t *data = (_simple_event_data_t *)_data;
+    _identity_t *domains = data->domains;
+    uint64 no_domains = data->no_domains;
+
+    _identity_t match_id = *(_identity_t *)match_data;
+    for (uint64 i = 0; i < no_domains; ++i) {
+        if (_identity_eq(match_id, domains[i])) {
+            return 1;
+        }
+    }
+    return 0;
+}
+
+typedef enum {
+    _Simple_Event_Data_Format_Invalid,
+    _Simple_Event_Data_Format_Pseudodict,
+    _Simple_Event_Data_Format_Legacy
+} _simple_event_data_format_t;
+
+UNUSED static _simple_event_data_format_t
+_simple_event_data_representation(
+    attr_value_t val, bool has_indices, bool has_params) {
+    ASSERT(SIM_attr_is_list(val) && SIM_attr_list_size(val) > 0);
+    if (SIM_attr_list_size(val) == 2 && has_indices && has_params) {
+        // Either the pseudodict format, or the [indices, params] format
+        // Determine by inspecting the first element:
+        // [integer, ...] -> Legacy representation
```

Comment:
I think the two formats deserve a descriptive comment somewhere. In particular, specify the format of the legacy representation, and why it exists (compatibility with checkpoints older than 6.0.x). This way, it will be visible when it's clearly irrelevant and can be dropped.